### ISSUE TITLE
Custom prompt templates disregarded

### ISSUE THREAD

**Issue by jcbrabelo** [jerryjliu/gpt_index#596]
Custom prompt templates are disregarded by LlamaIndex. Steps to reproduce:

```
documents = SimpleDirectoryReader(<DOCS_PATH>).load_data()
llm_predictor = <...>
index = GPTSimpleVectorIndex(
    documents, llm_predictor=llm_predictor, text_qa_template=<CUSTOM_PROMPT>
)
response = index.query(question)
```

CUSTOM_PROMPT isn't applied to the query. 

I seems `CUSTOM_PROMPT` is present at the `index` object, but then it isn't sent along with the other query parameters when the query is actually executed.

A simple fix would be to include it as a keyword argument on the `query` method. If that is the intended usage, I suggest updating the documentation accordingly and removing it as a parameter of the index constructor.

**Comment by jerryjliu** [comment]
hey @jcbrabelo you have to specify it during query-time, not during index construction time (i get that this is confusing)

**Comment by jcbrabelo** [comment]
Thanks for replying, @jerryjliu . 

My point is that the documentation on https://gpt-index.readthedocs.io/en/latest/reference/indices/vector_store.html (and the code, too) accepts text_qa_template as a parameter but ends up not using it down the road. Also, the only custom prompt example I could find (https://gpt-index.readthedocs.io/en/latest/how_to/custom_prompts.html) sends the custom prompt parameter at index creation time.

Maybe defining a custom summarization prompt at a GPTTreeIndex creation time makes sense as it will be used to generate node summaries? Is there a reason why there is a text_qa_template parameter in the GPTSimpleVectorIndex constructor?

**Comment by jerryjliu** [comment]
Yeah you're right, i'll take a stab at improving this UX. in the meantime can you try specifying the templates during query-time?

**Comment by kenn** [comment]
The conversation here clarifies a lot. Thanks @jerryjliu and @jcbrabelo !

**Comment by jcbrabelo** [comment]
Absolutely @jerryjliu.
Thanks for looking into it!

**Comment by jerryjliu** [comment]
This should be fixed in #639 (so you can also define during index construction)
